src/rgw/rgw_sal_motr.cc

```diff
@@ -2146,30 +2171,62 @@ int MotrObject::read_mobj(const DoutPrefixProvider* dpp, int64_t off, int64_t en
 
 int MotrObject::get_bucket_dir_ent(const DoutPrefixProvider *dpp, rgw_bucket_dir_entry& ent)
 {
-  int rc = 0;
+  int rc = 0,ret_code;
   string tenant_bkt_name = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
   string bucket_index_iname = "motr.rgw.bucket.index." + tenant_bkt_name;
-  int max = 1000;
+  int max = 2;
   vector<string> keys(max);
   vector<bufferlist> vals(max);
   bufferlist bl;
   bufferlist::const_iterator iter;
 
+  // get an extra null entry
+  std::string null_version_key = this->get_name() + "/null";
```

Comment:
We should lookup null object only when the key.instance="null". Move this lookup to `if(this->get_key().have_null_instance())` block.

Reply:
done the changes as suggested.